src/OmniSharp.Roslyn.CSharp/Services/Navigation/GotoDefinitionService.cs

```diff
@@ -31,8 +31,12 @@ public GotoDefinitionService(OmniSharpWorkspace workspace, MetadataHelper metada
         {
             var quickFixes = new List<QuickFix>();
 
-            var document = _workspace.GetDocument(request.FileName);
+            var document = request.IsMetadataFile() ? 
+                _metadataHelper.FindDocumentInMetadataCache(request.FileName) : 
+                _workspace.GetDocument(request.FileName);
```

Comment:
We could probably just make it `_metadataHelper.FindDocumentInMetadataCache(request.FileName) ?? _workspace.GetDocument(request.FileName);` to avoid the first if there.